libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -955,23 +991,37 @@ std::string IRGenerator::deployCode(ContractDefinition const& _contract)
 	if (_contract.isLibrary())
 	{
 		solAssert(ContractType(_contract).immutableVariables().empty(), "");
-		immutables.emplace_back(std::map<std::string, std::string>{
-			{"immutableName"s, IRNames::libraryAddressImmutable()},
-			{"value"s, "address()"}
-		});
-
+		if (!eof)
+			immutables.emplace_back(std::map<std::string, std::string>{
+				{"immutableName"s, IRNames::libraryAddressImmutable()},
+				{"value"s, "address()"}
+			});
+		else
+			t("libraryAddressImmutableOffset", std::to_string(m_context.libraryAddressImmutableOffset()));
 	}
 	else
+	{
 		for (VariableDeclaration const* immutable: ContractType(_contract).immutableVariables())
 		{
 			solUnimplementedAssert(immutable->type()->isValueType());
 			solUnimplementedAssert(immutable->type()->sizeOnStack() == 1);
-			immutables.emplace_back(std::map<std::string, std::string>{
-				{"immutableName"s, std::to_string(immutable->id())},
-				{"value"s, "mload(" + std::to_string(m_context.immutableMemoryOffset(*immutable)) + ")"}
-			});
+			if (!eof)
+				immutables.emplace_back(std::map<std::string, std::string>{
+					{"immutableName"s, std::to_string(immutable->id())},
+					{"value"s, "mload(" + std::to_string(m_context.immutableMemoryOffset(*immutable)) + ")"}
+				});
 		}
-	t("immutables", std::move(immutables));
+	}
+
+	if (eof)
+	{
+		t("auxDataStart", std::to_string(CompilerUtils::generalPurposeMemoryStart));
+		yulAssert(m_context.reservedMemorySize() <= 0xFFFF, "Reserved memory size exceeded EOF data section size.");
```

Comment:
```suggestion
		solAssert(m_context.reservedMemorySize() <= 0xFFFF, "Reserved memory size exceeded maximum allowed EOF data section size.");
```

TBH, we should just replace them all with a single assertion macro. But for now we do have this weird distinction and this is not a part of `libyul`.

Reply:
changed.